pagopa/io-app
# [IA-258] Add patch react-native-webview (Android)

## Short description
This PR adds a patch to [react-native-webview](https://github.com/react-native-webview/react-native-webview) that introduces two new property that works only for android

```typescript
/**
   * Boolean value to always forbid access to the microphone in the `WebView`, even if the app
   * was granted the necessary Android permission. The default value is `false` for backward-compatibility.
   * @platform android
   */
  androidMicrophoneAccessDisabled?: boolean;

 /**
   * Boolean value to always forbid access to the camera in the `WebView`, even if the app
   * was granted the necessary Android permission. The default value is `false` for backward-compatibility.
   * @platform android
   */
  androidCameraAccessDisabled?: boolean;
```

their default value is `false` so it works as previous.
Using `true`, instead, you'll deny the accesso to the camera/mic to these sites that want to access

### how to test
- use an Android device
- `yarn postinstall` (to apply the patch)
- go to profile
- activate the dev mode
- tap on **_My Portal Web PlayGround_**
- insert this site to the the device camera https://webcamtests.com
- test this scenario

| result expected                         | instructions                                                                                                                                                                                                       |
|-----------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| the access to the camera is not allowed | - visit the site - try to access one of your listed cameras                                                                                                                                                        |
| the access to the camera is not allowed | - give the camera permission to the app: visit wallet/pay/manual insertion -> allow camera usage - go back into the web play ground - visit the site - try to access one of your listed cameras                    |
| the access to the camera is allowed     | - go to `ts/components/RegionServiceWebView.tsx` and turn to `true` the properties `androidMicrophoneAccessDisabled` and `androidCameraAccessDisabled` - visit the site - try to access one of your listed cameras |
